### PR TITLE
Chore: Exclude calls servers not in product telemetry

### PR DESCRIPTION
**Pull request by sri-byte**
Impact: Exclude instance in calls telemetry should be present in product telemetry, since calls is a feature of the product. This PR excludes servers that show up in calls telemetry and don't exist server telemetry.

### CODE REVIEW

**Review by ifoukarakis — CHANGES_REQUESTED** [review]

File: transform/mattermost-analytics/models/marts/product/calls/fct_calls_activity.sql [hunk 1]
```diff
@@ -8,4 +8,5 @@
 select
     {{dbt_utils.star(ref('int_calls_active_days_spined'))}}
 from
-    {{ ref('int_calls_active_days_spined') }}
\ No newline at end of file
+    {{ ref('int_calls_active_days_spined') }} ca
+    where exists (select 1 from {{ ref('int_server_hosting_type') }} si where ca.server_id = si.server_id)
```
Comment (ifoukarakis) [inline transform/mattermost-analytics/models/marts/product/calls/fct_calls_activity.sql:12]: Let's document why these servers are excluded.

**Review by ifoukarakis — COMMENTED** [review]
Only thing missing is documentation on which servers are excluded.

File: transform/mattermost-analytics/models/marts/product/calls/fct_calls_activity.sql [hunk 2]
```diff
@@ -8,4 +8,5 @@
 select
     {{dbt_utils.star(ref('int_calls_active_days_spined'))}}
 from
-    {{ ref('int_calls_active_days_spined') }}
\ No newline at end of file
+    {{ ref('int_calls_active_days_spined') }} ca
+    where exists (select 1 from {{ ref('dim_server_info') }} si on ca.server_id = si.server_id)
```
[the remark this answers is not in the document]
Comment (sri-byte) [reply transform/mattermost-analytics/models/marts/product/calls/fct_calls_activity.sql]: Thank you. I split the logic from `dim_server_info` into another intermediate model here https://github.com/mattermost/mattermost-data-warehouse/pull/1377 and this is now update to use the new model.

File: transform/mattermost-analytics/models/marts/product/calls/fct_calls_activity.sql (hunk 1 above)
Comment (sri-byte) [reply transform/mattermost-analytics/models/marts/product/calls/fct_calls_activity.sql:12]: This is now documented in Confluence.

File: transform/mattermost-analytics/models/marts/product/calls/fct_calls_activity.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (sri-byte) [reply transform/mattermost-analytics/models/marts/product/calls/fct_calls_activity.sql]: Updated.

**Review by ifoukarakis — APPROVED** [review]

**Review by catalintomai — APPROVED** [review]

**Pull request merged**